.github/workflows/test.yml

```diff
@@ -91,14 +97,15 @@ jobs:
           sudo mv packmol /usr/local/bin/
           cd ..
         continue-on-error: true # This is not critical to succeed.
+
       - name: Install dependencies
         run: |
-          python -m pip install numpy cython
+          uv pip install numpy cython --system
 
-          # TODO remove next line installing ase from main branch until FrechetCellFilter is released
-          pip install git+https://gitlab.com/ase/ase
+          uv pip install -e '.[dev,optional]' --system
```

Comment:
@janosh, sorry for digging out this. But why do we need to install "editable" for CI?

Reply:
i don't remember exactly. i'm fairly sure i tried non-editable first and it failed for some reason. but that was with an earlier `uv` and might have changed

Reply:
I could confirm [install "non-editable" indeed breaks the tests](https://github.com/DanielYang59/pymatgen/actions/runs/8628590384/job/23651200610). The `pip install` process seems to run without any issue, but `pytest` seems unable to access test files. I don't quite understand this behavior.

An off-topic question: do we really need: https://github.com/materialsproject/pymatgen/blob/7064c439cbd57d90a39af9659fb206d9eaf22154/.github/workflows/test.yml#L18-L21
By default workflows would not run on forks until enabled by fork owner. Having this would prevent fork from running workflows, which seems unnecessary (some people might prefer to test on their own fork before pushing to upstream)? Or in my case I have to remove this line to run tests on my own fork.

<img width="1437" alt="Screenshot 2024-04-10 at 16 57 58" src="https://github.com/materialsproject/pymatgen/assets/80093591/a766cc38-dcf6-430f-8987-e108f42d3cf8">

Reply:
editable install basically just symlinks the repo into `site-packages`, meaning the whole repo contents including test files are symlinked. i think with non-editable install `TEST_FILES_DIR` points at a non-existent directory.

> why do we need:
```yml
     if: github.repository == 'materialsproject/pymatgen' 
```

not sure if this changed but people used to get every CI failure email twice from GitHub, once on their fork and once on open PRs which was very annoying

Reply:
i'm guessing if anything, `--editable` offers a performance gain given how fast it is to symlink

> Otherwise people who want to run workflow on their fork for any reason would need to remove this line?

not sure that's an issue? people usually just fork to make PRs in which CI runs on this repo. if they want to maintain a fork long-term, removing this line is not much work?

Reply:
> i'm guessing if anything, `--editable` offers a performance gain given how fast it is to symlink

Not sure, but I would test later and let you know.

> people usually just fork to make PRs in which CI runs on this repo. 

Yes. But my point is... By default workflow would NOT run on forks (I assume people would enable workflow on their fork only when they have a good reason to), such that people could easily toggle workflow on and off easily on their fork (and have more control over their fork).

But if we disable workflow for them, they can not easily enable it. (I guess click a button would be easier than changing a line and make a commit? Also if they want to to push their change to upstream, they would have to revert it.)